web/src/redcross/blogModule/components/BlogComponent.ts

```diff
@@ -1,10 +1,45 @@
-import {Component} from "@angular/core";
+import {Component, OnInit, Inject} from "@angular/core";
+import {BlogPostComponent} from "./BlogPostComponent";
+import {BlogDialogComponent} from "./BlogDialogComponent";
+import {MatDialog, MatDialogRef, MAT_DIALOG_DATA} from "@angular/material";
+
+import {DiscussionDataService} from "./BlogDataService";
+
+import {Post} from "./Post";
+import {Author} from "./Author";
 
 @Component({
     selector: "blog",
     templateUrl: "./Blog.ng.html",
-    styleUrls: ["./Blog.less"]
+    styleUrls: ["./Blog.less"],
+    providers: [DiscussionDataService, MatDialog]
 })
 export class BlogComponent {
 
+    private posts: Post[];
+    private currentUser: Author;
+
+    constructor(public dds: DiscussionDataService, public md: MatDialog) {}
```

Comment:
these should be private. If you do intend to let other classes to use them, always make them private! Check other classes constructor too!